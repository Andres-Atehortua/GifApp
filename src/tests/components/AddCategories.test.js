import React from "react";
import { shallow } from "enzyme";
import AddCategory from "components/AddCategories";

describe("Pruebas al componente <AddCategories />", () => {
  const setCategories = jest.fn();

  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("should render correctly the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change the input text", () => {
    const input = wrapper.find("input[type='text']");
    const value = "Hola mundo";
    input.simulate("change", { target: { value } });
    const inputAfter = wrapper.find("input[type='text']").prop("value");

    expect(inputAfter).toBe(value);
  });

  test("should not post info when submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should call setCategories and clear the input value", () => {
    const value = "Hola mundo";
    const input = wrapper.find("input[type='text']");
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    expect(setCategories).toHaveBeenCalled();
    // Esto es para comprobar que se ha llamado con una funci√≥n
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(input.prop("value")).toBe("");
  });
});
