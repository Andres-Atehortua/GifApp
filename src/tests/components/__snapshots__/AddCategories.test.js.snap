// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <AddCategories /> should render correctly the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
